### PR TITLE
add an option to specify consul token

### PR DESCRIPTION
**Pull request by sbko**

### CODE REVIEW

File: cmd/istio-registry-sync/main.go [hunk 1]
```diff
@@ -145,33 +146,36 @@ func serve() (serve *cobra.Command) {
...
-	if awsErr == nil {
-		log.Infof("Cloud Map Watcher initialized in %q", awsRegion)
+	cmWatcher, err := cloudmap.NewWatcher(ctx, store, awsRegion, awsID, awsSecret)
+	if err != nil {
+		log.Errorf("error setting up aws: %v", err)
 	}
-	consulWatcher, consulErr := consul.NewWatcher(store, consulEndpoint, consulNamespace)
-	if consulErr == nil {
-		log.Infof("Consul Watcher initialized at %s", consulEndpoint)
+	consulWatcher, err := consul.NewWatcher(store, consulEndpoint, consulNamespace, consulToken)
```
Comment (liamawhite) [inline cmd/istio-registry-sync/main.go:162]: How is this actually used? I don't see any code?

Comment (sbko) [reply cmd/istio-registry-sync/main.go:162]: Line number 179 - https://github.com/tetratelabs/istio-registry-sync/pull/78/files#diff-0081bf84c038b3a9ac0e4fd9aa9d3695beae14cadfebaadabad90691c16b9f2fR179

Comment (liamawhite) [reply cmd/istio-registry-sync/main.go:162]: Sorry, I should have been more specific. I mean the token.

Comment (sbko) [reply cmd/istio-registry-sync/main.go:162]: https://github.com/tetratelabs/istio-registry-sync/pull/78/files#diff-c7f45d2f0713483a8ff992a91e1fa3d033b498d6c9610b4c649cb99dfff5afceR49

**Review by liamawhite — APPROVED** [review]

**Pull request merged**